src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -1464,6 +1505,11 @@ def stop_servers(self):
                 errors.append(
                     "ERROR: At least one multi-variant server was not found in "
                     "its expected state; stopping all servers")
+                # dump engines stacks if not already done
+                if self.dumped_engines_stacks is False:
+                    self.dumped_engines_stacks = True
+                    self.log.info("Some engine not in expected state, dumping ULT stacks")
+                    dump_engines_stacks(self.hostlist_servers)
```

Comment:
Per earlier suggestion
```suggestion
                # dump engines stacks if not already done
                self.dump_engine_stacks("Some engines not in expected state")
```